### PR TITLE
Fix for node.replace() bug

### PR DESCRIPTION
**Pull request by jywarren**
Attempted fix for https://github.com/publiclab/plots2/issues/2057

### CODE REVIEW

**Comment by siaw23** [comment]
yeah i did already. remaining one failure to be fixed. i posted in https://github.com/publiclab/plots2/issues/2057#issuecomment-360239575

**Comment by jywarren** [comment]
It's concerning that these errors did not occur in Travis, but if they work for you, I'm all right with it!

**Comment by siaw23** [comment]
all is fine now. if you merge I'll pull and continue work. i think there's something up with the setup with Travis. @jywarren

**Comment by siaw23** [comment]
Travis failed but I think this should be an easy fix.

**Comment by siaw23** [comment]
how about we specify ruby version on Travis? @jywarren

**Comment by jywarren** [comment]
Actually we do already! v2.3 -- hang on, trying another tweak here --

**Comment by siaw23** [comment]
OK. haven't checked Travis and setup though cause it I thought it's been taken care of. Let's hope this tweak fixes it...

**Comment by jywarren** [comment]
There's an issue with the ordering -- we hadn't set order for `revision.latest` but i just added that -- i think it was accounting for some problems where it wasn't sorting by timestamp... well, we'll see here; 

https://github.com/publiclab/plots2/blob/a701eb8b4c91c42a8311aa2922f66aaadf3e878b/test/unit/revision_test.rb#L42-L74

File: app/models/node.rb [hunk 1]
```diff
@@ -186,12 +186,13 @@ def likers
...
     revisions
       .where(status: 1)
+      .order(timestamp: :desc)
       .first
   end
 
   def revisions
     revision
-      .order('timestamp DESC')
+      .order(timestamp: :desc)
```
Comment (siaw23) [inline app/models/node.rb]: btw this is one of the changes the style guide takes care of. if we implement it will save us a lot of headache.

**Comment by jywarren** [comment]
There may also be a timestamp based issue, if these tests rely on timestamp ordering and the tests run in less than one second! But we can use Timecop to resolve this: 

https://github.com/publiclab/plots2/blob/a9ae22477ace5eb34c5e7883dbc48819b5d01637/test/functional/admin_controller_test.rb#L20

**Comment by jywarren** [comment]
Hope this works for you!

**Pull request merged**